src/main/scala/mesosphere/marathon/core/storage/simple/SimplePersistenceStore.scala

```diff
@@ -0,0 +1,155 @@
+      .groupBy(parallelism, node => Math.abs(node.path.hashCode) % parallelism)
+      .map(node => factory
+        .create()
+        .forPath(node.path, node.data.toArray)
+        .toScala)
+      .mapAsync(parallelism)(identity)
+      .mergeSubstreams
+
+  override def read: Flow[String, Try[Node], NotUsed] =
+    Flow[String]
+      .via(logPath("Reading a node at "))
+      .via(metric(readMetric))
+      .map { path =>
+        factory
+          .getData()
+          .forPath(path)
+          .toScala
+          .map(bytes => Try(Node(path, ByteString(bytes))))
+          .recover {
+            case e: NoNodeException => Failure(e) // re-throw exception in all other cases
+          }
+      }
+      .mapAsync(parallelism)(identity)
+
+  override def update: Flow[Node, String, NotUsed] =
+    Flow[Node]
+      .via(logNode("Updating a node at "))
+      .via(metric(updatedMetric))
+      .groupBy(parallelism, node => Math.abs(node.path.hashCode) % parallelism)
+      .map(node => factory
+        .setData()
+        .forPath(node.path, node.data.toArray)
+        .toScala
+        .map(_ => node.path))
+      .mapAsync(parallelism)(identity)
+      .mergeSubstreams
+
+  override def delete: Flow[String, String, NotUsed] =
+    Flow[String]
+      .via(logPath("Deleting a node at "))
+      .via(metric(deleteMetric))
+      .map(path => factory
+        .delete()
+        .forPath(path)
+        .toScala
+        .map(_ => path)) // Note: A Void is returned by the delete call and it can not be passed to the mapAsync stage
+      .mapAsync(parallelism)(identity)
+
+  override def children(path: String): Future[Seq[String]] = {
+    childrenMetric.increment()
+    logger.debug(s"Getting children at $path")
+    factory
+      .children()
+      .forPath(path).toScala
+      .map(JavaConverters.asScalaBuffer(_).to[Seq])
+  }
+
+  override def exists(path: String): Future[Boolean] = {
+    existsMetric.increment()
+    logger.debug(s"Checking node existence for $path")
+    factory
+      .checkExists()
+      .forPath(path).toScala
+      .map(stat => if (stat == null) false else true)
+  }
+
+  override def sync(path: String): Future[Done] = {
+    logger.debug(s"Syncing nodes for path $path")
+    factory
+      .sync()
+      .forPath(path).toScala
+      .map(_ => Done)
+  }
+
+  override def transaction(operations: Seq[StoreOp]): Future[Done] = {
+    logger.debug(s"Submitting a transaction with ${operations.size} operations")
+    transactionCountMetric.increment(operations.size.toLong)
+    transactionMetric.increment()
+
+    val transactionOps = operations.map {
+      case CreateOp(node) => factory.transactionOpCreate().forPath(node.path, node.data.toArray)
+      case UpdateOp(node) => factory.transactionOpSetData().forPath(node.path, node.data.toArray)
+      case DeleteOp(path) => factory.transactionOpDelete().forPath(path)
+      case CheckOp(path) => factory.transactionOpCheck().forPath(path)
```

Comment:
What is `CheckOp` doing? Is it of value if we return `Done`?

Reply:
`CheckOp` checks if a node with given path exists. See [this](https://github.com/mesosphere/marathon/pull/6265/files#diff-d6ddb78bb38bf0c43289c466db30259fR350) test.

Reply:
I am just kind of alergic to the `Op` suffix, can't we do just Create, Update, Delete etc.?

Also can't we name CheckOp -> Exists or CheckExists?

Reply:
So far wrappers are called the same as their Curator counterparts. I understand the "allergy" but `Op` here actually makes sense signifying that it's not a `Create` operation but rather a `transactionOpCreate`. Unless you have a better proposal I'm inclined to leave it like this ;)